### ISSUE TITLE
setstatus  should fail if nothing is done

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#372]
In GitLab by @etiennesky on Feb 26, 2019, 17:34

Hi @dbeltrankyl sometimes I make a typo in the setstatus command which results in nothing being done.

No error is printed, and the autosubmit command runs without doing anything.

This is a bug and should be fixed.

e.g. 

```
etourign@ bscesautosubmit02:/esarchive/autosubmit/a1r9/proj/auto-ecearth3> autosubmit setstatus -ft a1r9_19010101_fc0_1_POST  -t READY -s -np a1r9
Preparing .lock file to avoid multiple instances with same expid.

Checking configuration files...
platforms_a1r9.conf OK
jobs_a1r9.conf OK
autosubmit_a1r9.conf OK
expdef_a1r9.conf  OK
Configuration files OK

Creating jobs...
Loading JobList
Adding dependencies...
Removing redundant dependencies...
etourign@ bscesautosubmit02:/esarchive/autosubmit/a1r9/proj/auto-ecearth3> autosubmit setstatus -fl a1r9_19010101_fc0_1_POST  -t READY -s -np a1r9
Preparing .lock file to avoid multiple instances with same expid.

Checking configuration files...
platforms_a1r9.conf OK
jobs_a1r9.conf OK
autosubmit_a1r9.conf OK
expdef_a1r9.conf  OK
Configuration files OK

Creating jobs...
Loading JobList
Adding dependencies...
Removing redundant dependencies...
CHANGED: job: a1r9_19010101_fc0_1_POST status to: READY
etourign@ bscesautosubmit02:/esarchive/autosubmit/a1r9/proj/auto-ecearth3> 

```

**Comment by kinow** [comment]
In GitLab by @wuruchi on Jul 16, 2019, 11:49

I will start working on this. At first glance at the code, seems quite approachable.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Jul 16, 2019, 17:06

mentioned in commit ac0e09c4f4a1bee17333699b7e117cbb4ab1084e

**Comment by kinow** [comment]
In GitLab by @wuruchi on Jul 31, 2019, 11:14

mentioned in commit d97b73cba54efdc3aaad1d1cb2810ea57519ed16

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 1, 2019, 13:17

closed via merge request !135

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 1, 2019, 13:17

mentioned in commit 808a14a337f0fbb7a82a7819e8ae1a12a0255240

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 1, 2019, 13:17

closed via commit ac0e09c4f4a1bee17333699b7e117cbb4ab1084e

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 1, 2019, 15:04

Hello team,

This feature has been implemented in !135.

Now setstatus should output proper validation information when setstatus receives incorrect input for its options -fc -ft -ftc -fs.

**Comment by kinow** [comment]
In GitLab by @etiennesky on Aug 1, 2019, 15:18

great! does AS also return a non-zero exit code so that the second statement (autosubmit run) is not executed?

`autosubmit setstatus -ft BLABLA -t READY -s -np $expid && autosubmit run $expid`

**Comment by kinow** [comment]
In GitLab by @wuruchi on Aug 1, 2019, 15:30

Hi @etiennesky 

We have tested that all possible combinations of -fc -ft -fs -ftc and their respective expected input return an error when required. So, the call `autosubmit run $expid` should not be executed.

Also, the validation will try to provide some intuition to the user as to what could have gone wrong.

Anyways, feel free to provide feedback. Expect to see these improvements in v3\.12.1b.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 1, 2019, 15:36

hi @etiennesky ,

Just tested this will be the output: 

`autosubmit setstatus -ft BLABLA -t READY -s -np a257 && autosubmit run a257`

```

## Section Validation Message ##
        Specified section(s) : [['BLABLA']] not found in the experiment a257.
        Process stopped. Review the format of the provided input. Comparison is case sensitive.
        Remember that this option expects section names separated by a blank space as input.
[CRITICAL] Error in the supplied input for -ft.
```

run in fact does not continue.

**Comment by kinow** [comment]
In GitLab by @etiennesky on Aug 1, 2019, 15:39

excellent! thanks a lot! please let me know when this change is released in the modules!

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 1, 2019, 15:43

It is already pushed into the modules in autosubmit/3.12.1b, changes like these won't be released into 3.12.0b since we decided a few weeks ago to make it a stable release so only will receive critical fixes.
